tlatools/org.lamport.tlatools/src/tlc2/tool/impl/SpecProcessor.java

```diff
@@ -1447,7 +1448,55 @@ private final void processConfigProps(String name, ExprNode pred, Context c, Lis
                 }
                 return;
             }
+
+			if (val instanceof OpDefNode) {
+				// Handle parameterized instantiation:
+				//
+				//   SM(param) == INSTANCE SomeModule WITH foo <- param
+				//   Property == ... SM(42)!Spec
+				//
+				// If pred1 is the application of a non-recursive, non-builtin, non-zero arity
+				// operator definition, the args define the values of the defs (formal)
+				// parameters. Thus, link the def's parameters to the args via the def's context.
+				// Afterwards, recurse on this operator definition.
+				// (Built-ins have no body, and I do not know how to handle recursive).
+				final OpDefNode odn = (OpDefNode) val;
+				if (odn.getBody() != null && !odn.getInRecursive() && odn.getArity() == args.length) {
+					// Link the operator's (formal) parameters to the given args.
+					c = symbolNodeValueLookupProvider.getOpContext(odn, args, c, false, toolId);
+					// Recurse.
+					this.processConfigProps(opNode.getName().toString(), odn.getBody(), c, subs);
+					return;
+				}
+			}
+          
             int opcode = BuiltInOPs.getOpCode(pred1.getOperator().getName());
+			if (opcode == OPCODE_bf) {
+				// Handle universal quantification in parameterized instantiation:
+				//
+				//   SM(param) == INSTANCE SomeModule WITH foo <- param
+				//   Property == \A n \in 1..4 : SM(n)!Spec
+				//
+				try {
+					final ContextEnumerator ctxts = opDefEvaluator.contexts(pred1, c);
+					if (ctxts.isDone()) {
+						Assert.fail(EC.TLC_LIVE_FORMULA_TAUTOLOGY);
+					}
+					Context c1;
+					while ((c1 = ctxts.nextElement()) != null) {
+						final ExprNode expr = (ExprNode) args[0];
+						this.processConfigProps(expr.toString(), expr, c1, subs);
+					}
+					return;
+				} catch (Exception ignored) {
```

Comment:
TODO: Swallowing the exception risks that the `\A s \in ...` is only partially unrolled, leaving the property incomplete.